Reworks the migration sections to better showcase Hypershift (#2230)
timescale/docs: _partials/_hypershift-intro.md

You can use hypershift to migrate existing PostgreSQL databases in one step, and
enable compression and create hypertables instantly.

Use Hypershift to migrate your data to Timescale from these sources:

*   Standard PostgreSQL databases
*   Amazon RDS databases
*   Other Timescale databases, including Managed Service for TimescaleDB, and
    self-hosted TimescaleDB

> Because compression is enabled during the migration, the maximum amount of 
> storage is not required in the target database before you start the migration.

Author: ```suggestion
```

You can use hypershift to migrate existing PostgreSQL databases in one step, and

> enable compression and hypertable creation on the fly.

Reviewer: ```suggestion
enable compression and create hypertable instantly.
```

> Hypershift can migrate your data in Timescale Cloud from these sources:

Reviewer: ```suggestion
Use Hypershift to migrate your data to Timescale from these sources:
```

*   Standard PostgreSQL databases
*   Amazon RDS databases
*   Other Timescale databases, including Managed Service for TimescaleDB, and
    self-hosted TimescaleDB

> Because compression is enabled during the migration, you do not need to have the
> maximum amount of storage available in the target database before you start
> migration.

Reviewer: ```suggestion
Because compression is enabled during the migration, a maximum amount of storage is not required in the target database before you start the migration.
```
